backend/errors/error_source.go

```diff
@@ -0,0 +1,37 @@
+package errors
+
+import (
+	"net/http"
+)
+
+func GetErrorSource(statusCode int) ErrorSource {
+	switch statusCode {
+	case http.StatusBadRequest,
```

Comment:
Nit. Seems like this can simplified by checking >= 500 like we do here https://github.com/grafana/grafana/blob/a2d4ce18adafbd79380a7feb6e3f5c80049cc2bb/pkg/middleware/requestmeta/request_metadata.go#L109

Reply:
I don't think all 400 errors are caused by the plugin.  Maybe opinionated, but I explicitly listed the 400 errors that can be caused by the plugin.  Things like 403 are probably an issue with plugin configuration.

Reply:
It's a bit clearer now why we defined a third source "external" in core grafana.  That could indicate an error outside the plugin, but not necessarily "downstream", like a problem with user configuration, which seems more "upstream".

Reply:
Good points. I think its mainly like that because Grafana operators currently only look at 5xx statuses when defining SLO's. 

> It's a bit clearer now why we defined a third source "[external](https://github.com/grafana/grafana/blob/main/pkg/tsdb/prometheus/instrumentation/instrumentation.go#L25)" in core grafana. That could indicate an error outside the plugin, but not necessarily "downstream". For example, a problem with user configuration, which seems more "upstream".

I understand what you mean, but I guess there's a question whether there's any use of this to define SLO's. I would think no. Maybe you have other use cases in mind?

Reply:
From an SLO standpoint seems unnecessary.  But to show a dashboard of "downstream" vs "upstream" errors would be nice.